frontend/src/component/common/NewConstraintAccordion/ConstraintsList/EditableConstraintsList.tsx

```diff
@@ -1,24 +1,23 @@
+import type React from 'react';
 import { forwardRef } from 'react';
 import { styled } from '@mui/material';
 import type { IConstraint } from 'interfaces/strategy';
 import produce from 'immer';
 import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
-import { useWeakMap } from 'hooks/useWeakMap';
 import { constraintId } from 'component/common/LegacyConstraintAccordion/ConstraintAccordionList/createEmptyConstraint';
 import { ConstraintsList } from 'component/common/ConstraintsList/ConstraintsList';
 import { EditableConstraintWrapper } from 'component/feature/FeatureStrategy/FeatureStrategyConstraints/EditableConstraintWrapper';
-import type {
-    IConstraintsListProps as IEditableConstraintsListProps,
-    IConstraintsListRef as IEditableConstraintsListRef,
-    IConstraintsListItemState as IEditableConstraintsListItemState,
-} from './ConstraintsListUtils';
-import { useConstraintsList } from './ConstraintsListUtils';
 
-export type { IEditableConstraintsListProps, IEditableConstraintsListRef };
+export interface IEditableConstraintsListRef {
+    addConstraint?: (contextName: string) => void;
+}
 
-export const editableConstraintsListId = 'editableConstraintsListId';
+export interface IEditableConstraintsListProps {
+    constraints: IConstraint[];
+    setConstraints?: React.Dispatch<React.SetStateAction<IConstraint[]>>;
+}
 
-export const useEditableConstraintsList = useConstraintsList;
+export const editableConstraintsListId = 'editableConstraintsListId';
```

Comment:
Can we use react's `useId` hook instead? So we never need to worry about having crashing IDs on a page
```suggestion
export const editableConstraintsListId = useId();
```